frontend/src/component/Account/UserSpacesConfig.jsx

```diff
@@ -0,0 +1,135 @@
+            </td>
+
+            <td>
+              <div className='spaceconfig__table__role'>
+                <div className='spaceconfig__table__role__icon'>
+                  <i className={`fa fa-fw fa-${memberRole.faIcon}`} style={{ color: memberRole.hexcolor }} />
+                </div>
+                <div className='spaceconfig__table__role__text d-none d-sm-flex'>
+                  {props.t(memberRole.label)}
+                </div>
+              </div>
+            </td>
+
+            <td>
+              <BtnSwitch
+                checked={member.doNotify}
+                onChange={() => props.onChangeSubscriptionNotif(space.id, !member.doNotify)}
+              />
+            </td>
+            <td>
+              <IconButton
+                className='outlineTextBtn primaryColorBorder primaryColorBgHover primaryColorBorderDarkenHover'
+                onClick={() => this.setState({ spaceBeingDeleted: space.id })}
+                icon='sign-out'
+                text={props.admin ? props.t('Remove from space') : props.t('Leave space')}
+              />
+            </td>
+          </tr>
+        )
+      }
+    }).filter(entry => entry)
+
+    return (
+      <div className='account__userpreference__setting__spacename'>
+        <div className='spaceconfig__sectiontitle subTitle ml-2 ml-sm-0'>
+          {props.t('Spaces')}
+        </div>
+
+        <div className='spaceconfig__text ml-2 ml-sm-0' />
+
+        {(entries.length ? (
+          <div className='spaceconfig__table'>
+            <table className='table'>
+              <thead>
+                <tr>
+                  <th>{props.t('Space')}</th>
+                  <th>{props.t('Role')}</th>
+                  <th>{props.t('Email notifications')}</th>
+                  <th />
+                </tr>
+              </thead>
+
+              <tbody>{entries}</tbody>
+            </table>
+            {(this.state.spaceBeingDeleted && (
+              <ConfirmPopup
+                onConfirm={this.handleConfirmDeleteSpace}
+                onCancel={() => this.setState({ spaceBeingDeleted: null })}
+                msg={
+                  props.admin
+                    ? props.t('Are you sure you want to remove this member from the space?')
+                    : props.t('Are you sure you want to leave the space?')
+                }
+                confirmLabel={
+                  props.admin
+                    ? props.t('Remove from space')
+                    : props.t('Leave space')
+                }
+              />
+            ))}
+          </div>
+        ) : (
+          props.admin
+            ? props.t('This user is not a member of any space yet')
+            : props.t('You are not a member of any space yet')
+        ))}
+      </div>
+    )
+  }
+}
+
+const mapStateToProps = ({ workspaceList }) => ({ workspaceList })
+export default connect(mapStateToProps)(translate()(UserSpacesConfig))
+
+UserSpacesConfig.propTypes = {
```

Comment:
userToEditId and admin props aren't marked as `required` but don't have any default values

Reply:
Indeed, I didn't know `isRequired`. Fixed.